src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -344,6 +346,7 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     if (_options.Adapters.Any(adapter => adapter.NeedsRefresh()) || changedKeyValuesCollection?.Any() == true || keyValueChanges.Any())
                     {
                         _applicationSettings = applicationSettings;
+                        _watchedSettings = watchedSettings;

```

Comment:
Where are we updating the watched setting for individual keys?

Reply:
From what I'm seeing, it may have been a bug introduced when the preview branch was created. But I'm not sure if I'm just missing it.

Reply:
Its being updated in the loop right after this.